app/code/community/Paylike/Paymentmethod/Model/Paymentmethod.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+require_once dirname(__FILE__) . '/../Model/Paylike.php';
+class Paylike_Paymentmethod_Model_Paymentmethod extends Mage_Payment_Model_Method_Cc
+{
+    const REQUEST_TYPE_AUTH_CAPTURE = 'AUTH_CAPTURE';
+    const REQUEST_TYPE_AUTH_ONLY = 'AUTH_ONLY';
+
+    protected $_isGateway = true;
+    protected $_canAuthorize = true;
+    protected $_canUseCheckout = true;
+    protected $_canCapture = true;
+    protected $_canCapturePartial = false;
+    protected $_canRefund = true;
+    protected $_canRefundInvoicePartial = true;
+    protected $_canVoid = true;
+    protected $_code = 'paymentmethod';
+    protected $_formBlockType = 'paymentmethod/form_paymentmethod';
+    protected $_infoBlockType = 'paymentmethod/info_paymentmethod';
+
+    public function authorize(Varien_Object $payment, $amount)
+    {
+        $payment->setTransactionId($payment->getPaylikeTransactionId());
+        $payment->setIsTransactionClosed(0);
+        return $this;
```

Comment:
I think you should verify the transaction here.
On each transaction you should verify that the transaction belongs to the current order, and has the expected currency and amount. If not, the transaction has been manipulated and the payment should fail.
It is as simple as fetching the transaction from Paylike and matching the data with what you have on record.